pkg/operator/controller/extension/extension/runtime/runtime.go

```diff
@@ -46,27 +51,37 @@ type deployer struct {
 }
 
 // Reconcile creates or updates the extension deployment in the garden runtime cluster.
-func (d *deployer) Reconcile(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) error {
+func (d *deployer) Reconcile(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) (reconcile.Result, error) {
 	if !extensionDeploymentSpecified(extension) {
 		return d.Delete(ctx, log, extension)
 	}
 
+	if err := d.runtimeClientSet.Client().Get(ctx, client.ObjectKey{Name: priorityClassName}, &schedulingv1.PriorityClass{}); err != nil {
```

Comment:
I assume this can also be true for essential CRDs like `ManagedResource`? Should it be checked here, as well?

Reply:
As discussed offline, waiting for the `ManagedResource` CRD happens implicitly by waiting for the `PriorityClass`.